gtsfm/averaging/rotation/cycle_consistency.py

```diff
@@ -1,343 +0,0 @@
-        per_edge_errors[(i0, i1)].append(cycle_error)
-        per_edge_errors[(i1, i2)].append(cycle_error)
-        per_edge_errors[(i0, i2)].append(cycle_error)
-
-        cycle_errors.append(cycle_error)
-        max_rot_errors.append(max_rot_error)
-
-    inlier_errors_aggregate = []
-    inlier_errors_wrt_gt = []
-
-    outlier_errors_aggregate = []
-    outlier_errors_wrt_gt = []
-
-    plt.close("all")
-    # aggregate info over per edge_errors
-    for (i1, i2), edge_cycle_errors in per_edge_errors.items():
-        if edge_acceptance_criterion == EdgeErrorAggregationCriterion.MIN_EDGE_ERROR:
-            error_aggregate = np.amin(edge_cycle_errors)
-
-        elif edge_acceptance_criterion == EdgeErrorAggregationCriterion.MEDIAN_EDGE_ERROR:
-            error_aggregate = np.median(edge_cycle_errors)
-
-        if error_aggregate < CYCLE_ERROR_THRESHOLD:
-            cycle_consistent_keys.add((i1, i2))
-            inlier_errors_aggregate.append(error_aggregate)
-            inlier_errors_wrt_gt.append(two_view_reports_dict[(i1, i2)].R_error_deg)
-        else:
-            outlier_errors_aggregate.append(error_aggregate)
-            outlier_errors_wrt_gt.append(two_view_reports_dict[(i1, i2)].R_error_deg)
-
-    if visualize:
-        plt.scatter(
-            inlier_errors_aggregate,
-            inlier_errors_wrt_gt,
-            10,
-            color="g",
-            marker=".",
-            label=f"inliers @ {CYCLE_ERROR_THRESHOLD} deg.",
-        )
-        plt.scatter(
-            outlier_errors_aggregate,
-            outlier_errors_wrt_gt,
-            10,
-            color="r",
-            marker=".",
-            label=f"outliers @ {CYCLE_ERROR_THRESHOLD} deg.",
-        )
-        plt.xlabel(f"{edge_acceptance_criterion} cycle error")
-        plt.ylabel("Rotation error w.r.t GT")
-        plt.axis("equal")
-        plt.legend(loc="lower right")
-        plt.savefig(os.path.join("plots", f"gt_err_vs_{edge_acceptance_criterion}_agg_error.jpg"), dpi=400)
-        plt.close("all")
-
-        plt.scatter(cycle_errors, max_rot_errors)
-        plt.xlabel("Cycle error")
-        plt.ylabel("Avg. Rot3 error over cycle triplet")
-        plt.axis("equal")
-        plt.savefig(os.path.join("plots", "cycle_error_vs_GT_rot_error.jpg"), dpi=400)
-        plt.close("all")
-
-    logger.info("cycle_consistent_keys: " + str(cycle_consistent_keys))
-
-    i2Ri1_dict_consistent, i2Ui1_dict_consistent, v_corr_idxs_dict_consistent = {}, {}, {}
-    for (i1, i2) in cycle_consistent_keys:
-        i2Ri1_dict_consistent[(i1, i2)] = i2Ri1_dict[(i1, i2)]
-        i2Ui1_dict_consistent[(i1, i2)] = i2Ui1_dict[(i1, i2)]
-        v_corr_idxs_dict_consistent[(i1, i2)] = v_corr_idxs_dict[(i1, i2)]
-
-    n_valid_edges = len([i2Ri1 for (i1, i2), i2Ri1 in i2Ri1_dict.items() if i2Ri1 is not None])
-    logger.info("Found %d consistent rel. rotations from %d original edges.", len(i2Ri1_dict_consistent), n_valid_edges)
```

Comment:
it would be nice to still log this summary info @akshay-krishnan

Reply:
added it.